runtime/interpreter/value.go

```diff
@@ -19855,6 +19855,24 @@ func (v *StorageReferenceValue) GetMember(
 ) Value {
 	self := v.mustReferencedValue(interpreter, locationRange)
 
+	switch name {
+	case sema.ReferenceTypeDereferenceFunctionName:
+		return NewHostFunctionValue(
+			interpreter,
+			sema.ReferenceDereferenceFunctionType(v.BorrowedType),
+			func(invocation Invocation) Value {
+				return self.Transfer(
+					invocation.Interpreter,
+					invocation.LocationRange,
+					atree.Address{},
+					false,
+					nil,
+					nil,
+				)
+			},
+		)
```

Comment:
Maybe refactor this into a function and use it here and in `EphemeralReferenceValue`, so it is ensured that the same code is used in both values

Reply:
Done